src/ocispec/ocijsonspec.hpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef OCIJSONSPEC_HPP_
+#define OCIJSONSPEC_HPP_
+
+#include <zephyr/data/json.h>
+
+#include <aos/common/ocispec.hpp>
+
+static constexpr size_t cMaxDeviceTreeLength = 1024;
+
+struct HypervisorSpec {
+    char*   path;
+    char*   parameters[aos::oci::cMaxParamCount];
+    uint8_t paramsLen;
+};
+
+struct KernelSpec {
+    char*   path;
+    char*   parameters[aos::oci::cMaxParamCount];
+    uint8_t paramsLen;
+};
+
+struct HWConfigSpec {
+    char* devicetree;
+};
+
+struct VMSpec {
+    HypervisorSpec hypervisor;
+    KernelSpec     kernel;
+    HWConfigSpec   hwconfig;
+};
+
+struct RuntimeSpec {
+    char*  ociVersion;
+    VMSpec vm;
+};
+
+static const struct json_obj_descr HypervisorSpecJsonDescr[] = {
+    JSON_OBJ_DESCR_PRIM(struct HypervisorSpec, path, JSON_TOK_STRING),
```

Comment:
I wonder, can we use Aos structures here? probably with some small modification?

Reply:
unfortunately no